### ISSUE TITLE
Blade Syntax Highlighting Issues

### ISSUE THREAD

**Issue by 7twin** [onecentlin/laravel-blade-snippets-vscode#94]
First of all, I am not trying to devalue any work that went into this, just trying to spark discussion and possible solution finding. I can't possibly imagine how much work this has to be to maintain with all those issues coming up and am thankful for how long it worked and made vscode a viable go-to environment for laravel developers.

But I am just wondering if the syntax highlighting is getting any updates or new maintainers are known or could be found? it has been a year or more with multitude of duplicate issues across all vscode blade language plugins (not just this one, though they all share the same issues and this one at least has commit timestamps ranging back to months, instead of multiple years).

All these issues forced many to switch over to atom or the premium offering from jetbrains, which is a shame, since vscode has so much more to offer than both combined. I sadly can't contribute myself or I would've already tried to help or even fork, fix and then push it back to origin master.

Hope somebody can jump in on this and help in any way possible to get the laravel environment inside vscode to be a viable replacement to all the others again.

Since we can't just expect this to be done purely out of good heart - while many work full-time most likely; some sort of patreon could be considered too, as I am sure a multitude of people would more than happily support these that would attempt to fix it. (with some proven progress possibly, so it doesn't end up like so many recent projects that took money and never helped the actual cause)

**Comment by cjoftheweb** [comment]
I think I replied to you on my repository. I suggest converting the Atom language syntax (which is a simple CSON -> JSON conversion) and submitting a PR to this repository.

Some suggestions for making a successful PR:

- Provide tests. This is the main reason I didn't accept some PRs on my repo.
- Software quality starts and ends with every one of us.
- If you have a problem, you are responsible for coming up with a solution. In the open source world, every programmer is capable of fixing any problem. If you aren't willing to do that, why should anyone else?

**Comment by 7twin** [comment]
@cjoftheweb Yes, indeed, and the reason I posted to your repository today, was because when comparing the many, they had very similar issues (which got fixed in atom alternatives) and some time ago - one of these packages directly had a note / link to your repository as the blade language included, so I figured I might ask the actual creator or the one that ported it to vsCode (and everybody else used) about how to possibly make further steps into making it maintained again - exactly because I wanted to help getting it working again, as the switch to atom for a working blade language was more than rocky otherwise, since most of the plugins got archived or killed entirely with 404s.

Since now I know that it is "just" CSON => JSON (though the atom repos seem to have much more things that vsCode languages don't seem to have at all, possibly why the atom stuff works "better" or more accurate?..)

I can package it up and test if it works, if it does, look into possible PR to this project too, though as mentioned, I am not sure how compatible it would be (was there any changes? is it just a fork of your port?), as I have not nearly enough knowledge about language templates in vsCode.

Otherwise there doesn't seem much response to this issue from the maintainer either, nor the existing PRs, so it's also questionable if not a separate fork would be better after all.

**Comment by 7twin** [comment]
@cjoftheweb an update on this, sadly just converting the [new atom config](https://github.com/jawee/language-blade/blob/master/grammars/blade.cson), does not solve anything, maybe a more specific issue with how vsCode handles language syntax versus atom, for example in atom you can easily have `@if` inside `@section`, but in vsCode that breaks the entire syntax highlighting for the rest of the document.
